ssv/attester.go

```diff
@@ -172,28 +172,37 @@ func (r *AttesterRunner) expectedPostConsensusRootsAndDomain() ([]ssz.HashRoot,
 func (r *AttesterRunner) executeDuty(duty *types.Duty) error {
 	// TODO - waitOneThirdOrValidBlock
 
-	attData, ver, err := r.GetBeaconNode().GetAttestationData(duty.Slot, duty.CommitteeIndex)
-	if err != nil {
-		return errors.Wrap(err, "failed to get attestation data")
-	}
-
-	attDataByts, err := attData.MarshalSSZ()
-	if err != nil {
-		return errors.Wrap(err, "could not marshal attestation data")
-	}
+	attestationFetcher := attestationFetcher(r, duty)
 
-	input := &types.ConsensusData{
-		Duty:    *duty,
-		Version: ver,
-		DataSSZ: attDataByts,
-	}
-
-	if err := r.BaseRunner.decide(r, input); err != nil {
+	if err := r.BaseRunner.decide(r, attestationFetcher); err != nil {
```

Comment:
Why are we passing r to r.BaseRunner.decide? Can't it access r internally?

Reply:
same for 
`aggregator`
`proposer`
and more

Reply:
It seems to me that the BaseRunner can't access the runner internally. But, also, the `decide` function doesn't need the runner as far as I understand. So, I think this could be simplified to `r.BaseRunner.decide(fetcher)` too